R/molecule_rank_plot.R

```diff
@@ -88,27 +88,71 @@ EdgeRankPlot.data.frame <- function (
            y = "Number of molecules") +
       theme_minimal()
 
-  return(edgerank_plot)
+  return(cellrank_plot)
 }
 
-#' @rdname EdgeRankPlot
-#' @method EdgeRankPlot Seurat
+#' @rdname MoleculeRankPlot
+#' @method MoleculeRankPlot Seurat
 #' @concept plots
 #'
 #' @examples
 #' library(pixelatorR)
 #'
 #' # Plot edge ranks with Seurat object
+#' MoleculeRankPlot(seur_obj)
+#'
+#' @export
+#'
+MoleculeRankPlot.Seurat <- function (
+  object,
+  group_by = NULL,
+  ...
+) {
+
+  cellrank_plot <- MoleculeRankPlot(object[[]], group_by = group_by)
+  return(cellrank_plot)
+}
+
+#' Edge Rank Plot
+#'
+#' @description
+#' `r lifecycle::badge("deprecated")`
+#'
+#' The function has been replaced by \code{\link{MoleculeRankPlot}} and
+#' will be removed in a future release.
+#'
+#' @param object A Seurat object
+#' @param group_by A character specifying a column to group by
+#' @param ... Additional arguments to pass to \code{\link{MoleculeRankPlot}}
+#'
+#' @rdname EdgeRankPlot
+#' @concept plots
+#'
+#' @examples
+#' library(pixelatorR)
+#'
+#' # Load example data as a Seurat object
+#' pxl_file <- system.file("extdata/five_cells",
+#'                         "five_cells.pxl",
+#'                         package = "pixelatorR")
+#' seur_obj <- ReadMPX_Seurat(pxl_file)
+#' seur_obj
+#'
+#' # Plot edge ranks with data.frame
 #' EdgeRankPlot(seur_obj)
 #'
 #' @export
 #'
-EdgeRankPlot.Seurat <- function (
+EdgeRankPlot <- function (
```

Comment:
Yes! If you type `?EdgeRankPlot`, there's a badge marking it as deprecated. So you can still use it on a Seurat object, but it will simply call `MoleculeRankPlot` under the hood.

Reply:
It is already deprecated in the main branch but popped up here due to the rebase.